commands/up.go

```diff
@@ -67,14 +73,62 @@ func upHandler(cmd *cobra.Command, args []string) error {
 	if err := runBuild(cmd, args); err != nil {
 		return err
 	}
-	fmt.Println()
+
 	if !skipPush {
 		if err := runPush(cmd, args); err != nil {
 			return err
 		}
-		fmt.Println()
 	}
-	if !skipDeploy {
+
+	if watch {
+		if err := runDeploy(cmd, args); err != nil {
+			return err
+		}
+
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			return err
+		}
+		defer watcher.Close()
+
+		_ = filepath.Walk("./", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return watcher.Add(path)
+			}
+			return nil
+		})
+
+		logger := log.New(os.Stderr, "", log.LstdFlags)
+		logger.SetFlags(log.Ldate | log.Ltime | log.LUTC)
+
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return fmt.Errorf("watcher's Events channel is closed")
+				}
+
+				if event.Op == fsnotify.Write {
```

Comment:
I think you need to debounce these notifications to prevent two identical builds happening if two files changed at the same time.

See how @johnmccabe handled this - there is a very old PR that you should find from him.

Reply:
Made changes for debouncing the notifications.

Reply:
https://github.com/openfaas/faas-cli/pull/763/files